### PR TITLE
Desktop: Allow selecting what views should be available from the Layout button

### PR DESCRIPTION
**Pull request by SubodhDahal**
- Added following dropdown for selecting available layouts in the "Appearance" config:
    - Editor / Viewer / Split View
    - Editor / Viewer
    - Editor / Split View
    - Viewer / Split View
- Pressing the "Layout" button toggles only between the layouts configured above

For issue #1900

### CODE REVIEW

**Comment by tessus** [comment]
Thanks for the PR. I was just wondering, if it wouldn't be better to create a new menu item `View -> Available layouts` and 3 menu items which can be checked: `Editor`, `Viewer` or `Preview`, and `Split view`.
But that's just an idea. Let's see how @laurent22 wants this to be implemented.

File: ElectronClient/app/app.js [hunk 1]
```diff
@@ -121,19 +121,31 @@ class Application extends BaseApplication {
...
+						let paneOptions;
+						if (Setting.value('layout') === Setting.LAYOUT_EDITOR_VIEWER) {
+							paneOptions = ['editor', 'viewer'];
+						} else if (Setting.value('layout') === Setting.LAYOUT_EDITOR_SPLIT) {
+							paneOptions = ['editor', 'both'];
+						} else if (Setting.value('layout') === Setting.LAYOUT_VIEWER_SPLIT) {
+							paneOptions = ['viewer', 'both'];
+						} else {
+							paneOptions = ['editor', 'viewer', 'both'];
+						}
```
Comment (laurent22) [inline ElectronClient/app/app.js]: The reducer should only read and write the state, so you'd want to access `state.settings.layout` instead. Accessing the Setting.XXX constants is fine.

File: ReactNativeClient/lib/models/Setting.js [hunk 2]
```diff
@@ -232,6 +232,21 @@ class Setting extends BaseModel {
...
+				public: true,
+				appTypes: ['desktop'],
+				isEnum: true,
+				label: () => _('Available Layouts'),
+				section: 'appearance',
+				options: () => ({
+					[Setting.LAYOUT_ALL]: _('Editor / Viewer / Split View'),
+					[Setting.LAYOUT_EDITOR_VIEWER]: _('Editor / Viewer'),
+					[Setting.LAYOUT_EDITOR_SPLIT]: _('Editor / Split View'),
+					[Setting.LAYOUT_VIEWER_SPLIT]: _('Viewer / Split View'),
```
Comment (laurent22) [inline ReactNativeClient/lib/models/Setting.js]: Let's make the life of translators easier by removing duplicated strings, so please change the first one to `_('%s / %s / %s', _('Editor'), _('Viewer'), _('Split View'))` and the other in a similar way.

**Comment by laurent22** [comment]
Thanks for the PR @SubodhDahal, I've left a few comments.

Also I agree with @tessus, the option should be in the menu, under "View" => "Layouts". Alternatively, it's also ok to make the setting as `public: false` and leave everything as it is, with just the existing "Toggle layout" menu item (no need to add new menu items). It's up to you, - just as long as that setting doesn't show up in the config screen.

**Comment by laurent22** [comment]
Perfect, thanks a lot for the quick update @SubodhDahal!

**Pull request merged**